agents/redfish/fence_redfish.py

```diff
@@ -0,0 +1,152 @@
+#!@PYTHON@ -tt
+
+# Copyright (c) 2018 Dell Inc. or its subsidiaries. All Rights Reserved.
+
+# Fence agent for devices that support the Redfish API Specification.
+
+import sys
+import re
+import json
+import requests
+import atexit
+sys.path.append("@FENCEAGENTSLIBDIR@")
+
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+from fencing import *
+from fencing import fail_usage, run_delay
+
+def get_power_status(conn, options):
+    uri = options["--systems-uri"]
+    response = send_get_request(options, uri)
+    if response['ret'] is False:
+        fail_usage("Couldn't get power information")
+    data = response['data']
+    if data[u'PowerState'].strip() == "On":
+        return "on"
+    else:
+        return "off"
```

Comment:
Are there any other states? We only want to return off when it's completely off, and you can return "unknown" for the other states so it retries until it's actually off.

Reply:
Good point. There are 2 other states that are very brief (but still possible): PoweringOn and PoweringOff, and of course Off. I will update this code to return "off" only when the state is "Off".

Reply:
Done